### PR TITLE
feat: Tron compatibility

### PR DESCRIPTION
**Pull request by aurel-fr**

### CODE REVIEW

File: src/tron/provider.ts [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    signedTransaction: string | Promise<string>
+  ): Promise<TransactionResponse> {
+    signedTransaction = await signedTransaction;
+    const deser = parseTransaction(signedTransaction);
+    const {to, data, from, value} = deser;
+    // is this a send eth transaction?
+    if (to && from && (!data || data == '0x')) {
+      return this.sendTrx(from, to, value);
+    }
+    // TODO, smart contract calls, etc
```
Comment (zouguangxian) [inline src/tron/provider.ts:182]: What will be the smart contract calls here?

**Review by zouguangxian — APPROVED** [review]

**Comment by zouguangxian** [comment]
It is not necessary to merge back to the master branch in this repository. and it is not a good practice to merge changes to the master branch which will cause trouble when syncing changes from the upstream.

File: src/tron/provider.ts (hunk 1 above)
Comment (aurel-fr) [reply src/tron/provider.ts:182]: This logic is sending TRX to an external account. This was part of my work to get deterministic deployment compatibility for Tron. At the onset Hardhat-deploy will fund the deploy proxy account, so this logic was needed.
It is not currently in use as we stopped short the development work for deterministic deployment but I'm leaving this if we ever decide to resume that work.

Comment (aurel-fr) [reply src/tron/provider.ts:182]: Calls to views functions or transactions. All of those require specific handling in Tron. For now there only is the logic for contract creation and sending TRX which is enough for regular deployments with hardhat-deploys.

**Pull request closed without merging**